tsk/fs/hfs.c

```diff
@@ -1730,6 +1730,15 @@ hfs_cat_file_lookup(HFS_INFO * hfs, TSK_INUM_T inum, HFS_ENTRY * entry,
 }
 
 
+static uint8_t
+hfs_find_highest_inum_cb(HFS_INFO * hfs, int8_t level_type,
+    const hfs_btree_key_cat * cur_key,
+    TSK_OFF_T key_off, void *ptr)
+{
+  *((TSK_INUM_T*) ptr) = tsk_getu32(hfs->fs_info.endian, cur_key->parent_cnid);
```

Comment:
Since the goal of this to find the biggest, I'd feel safer if this check first verified that the new value was indeed bigger than what is already there before overwriting ptr. Traverse doesn't really promise what order the callback will be called in.